src/commonMain/kotlin/exchange.dydx.abacus/state/manager/StateManagerAdaptor.kt

```diff
@@ -1750,6 +1757,121 @@ open class StateManagerAdaptor(
         return null
     }
 
+    private fun triggerOrderPayload(triggerOrder: TriggerOrder, subaccountNumber: Int, marketId: String, size: Double): HumanReadablePlaceOrderPayload {
+        val clientId = Random.nextInt(0, Int.MAX_VALUE)
+        val type = triggerOrder.type?.rawValue ?: throw Exception("type is null")
+        val side = triggerOrder.side?.rawValue ?: throw Exception("side is null")
+
+        val price = triggerOrder.summary?.price ?: throw Exception("summary.price is null")
+        val triggerPrice = triggerOrder.price?.triggerPrice ?: throw Exception("triggerPrice is null")
+
+        val reduceOnly = true
+        val postOnly = false
+
+        val timeInForce = when (triggerOrder.type) {
+            OrderType.stopMarket, OrderType.takeProfitMarket -> "IOC"
+            OrderType.stopLimit, OrderType.takeProfitLimit -> "GTT"
+            else -> throw Exception("invalid triggerOrderType")
+        }
+
+        val execution = when (triggerOrder.type) {
+            OrderType.stopMarket, OrderType.takeProfitMarket -> "IOC"
+            OrderType.stopLimit, OrderType.takeProfitLimit -> "Default"
+            else -> throw Exception("invalid triggerOrderType")
+        }
+
+        val duration = GoodTil.duration(TradeInputGoodUntil(28.0, "D")) ?: throw Exception("invalid duration")
```

Comment:
so they'll always default to 28 days? can we make a const somewhere?

Reply:
yup - will always default - will make a const!